### PR TITLE
Added rpk-001 keyboard

### PR DESCRIPTION
**Pull request by itarze**
## Description

as the title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/phentech/rpk_001/config.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2023 JoyLee (@itarze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* Encoder */
+#define ENCODER_DIRECTION_FLIP
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/config.h:7]: Should never be set at keyboard level
```suggestion
```

File: keyboards/phentech/rpk_001/config.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+/* LED */
+#define LED_PIN_ON_STATE 0
+#define LED_CAPS_LOCK_PIN C15
+#define LED_SCROLL_LOCK_PIN C0
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/config.h:12]: Must be configured in info.json
```suggestion
```

File: keyboards/phentech/rpk_001/info.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        "bootmagic": true,
+        "rgb_matrix": true
+    },
+    "matrix_pins": {
+        "cols": ["A6", "A7", "C4", "C5", "B0", "B1", "B10", "B11", "B13", "B14", "B15", "C6", "C7", "C8", "C9"],
+        "rows": ["C3", "A0", "A3", "A4", "A5"]
+    },
+    "encoder": {
+        "rotary": [
+            {"pin_a": "A1", "pin_b": "A2"}
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/info.json:29]:
```suggestion
            {"pin_a": "A2", "pin_b": "A1"}
```

File: keyboards/phentech/rpk_001/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        _______, _______, _______,                             _______,                              _______, _______, _______, _______, _______, _______
+    ),
+
+    [3] = LAYOUT_68_ansi(
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                             _______,                              _______, _______, _______, _______, _______, _______
+    )
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/keymaps/default/keymap.c:51]:
```suggestion
    )
```

File: keyboards/phentech/rpk_001/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        _______, _______, _______,                             _______,                              _______, _______, _______, _______, _______, _______
+    ),
+
+    [3] = LAYOUT_68_ansi(
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______,          _______, _______, _______,
+        _______, _______, _______,                             _______,                              _______, _______, _______, _______, _______, _______
+    )
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/keymaps/via/keymap.c:51]:
```suggestion
    )
```

File: keyboards/phentech/rpk_001/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+/* encoder; start */
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/keymaps/via/keymap.c:60]:
```suggestion
    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

File: keyboards/phentech/rpk_001/readme.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+* Keyboard Maintainer: [JoyLee](https://github.com/itarze)
+* Hardware Supported: RPK-001 PCB
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rpk_001:default
+
+Flashing example for this keyboard:
+
+    make rpk_001:default:flash
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/readme.md:15]:
```suggestion
    make phentech/rpk_001:default

Flashing example for this keyboard:

    make phentech/rpk_001:default:flash
```

File: keyboards/phentech/rpk_001/rpk_001.c [hunk 8]
```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 JoyLee (@itarze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/phentech/rpk_001/rpk_001.c:4]:
```suggestion
#include "quantum.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/phentech/rpk_001/info.json [hunk 9]
```diff
@@ -0,0 +1,187 @@
+{
+    "manufacturer": "HENGCANG ELECTRONIC",
+    "keyboard_name": "RPK-001",
+    "maintainer": "JoyLee",
+    "url": "",
+    "processor": "WB32FQ95",
+    "bootloader": "wb32-dfu",
+    "usb": {
+        "device_version": "1.0.0",
+        "vid": "0x3555",
```
Comment (drashna) [inline keyboards/phentech/rpk_001/info.json:10]: This VID appears to be in use, and should be changded.

File: keyboards/phentech/rpk_001/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______,  _______, _______, _______,  _______,  _______, _______, _______,          _______,          _______,
+        _______,          RGB_TOG, RGB_MOD, RGB_RMOD, RGB_SPI, _______, _______,  _______,  _______, _______, _______,          _______, _______, _______,
+        _______, GU_TOGG, _______,                             _______,                              _______, _______, _______, _______, _______, _______
+    )
+};
+
+/* encoder; start */
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (drashna) [inline keyboards/phentech/rpk_001/keymaps/via/keymap.c:40]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by itarze** [comment]
Can Merge? Or is there anything that needs fixing?

**Review by drashna — APPROVED** [review]

File: keyboards/phentech/rpk_001/info.json [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+            { "flags": 1, "matrix": [4, 8],  "x": 144, "y": 64},
+            { "flags": 1, "matrix": [4, 9],  "x": 160, "y": 64},
+            { "flags": 4, "matrix": [4, 10], "x": 176, "y": 64},
+            { "flags": 1, "matrix": [4, 11], "x": 192, "y": 64},
+            { "flags": 1, "matrix": [4, 13], "x": 208, "y": 64},
+            { "flags": 1, "matrix": [4, 14], "x": 224, "y": 64}
+        ]
+    },
+    "layouts": {
+        "LAYOUT_68_ansi": {
```
Comment (dunk2k) [inline keyboards/phentech/rpk_001/info.json:114]:
```suggestion
        "LAYOUT": {
```

File: keyboards/phentech/rpk_001/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    [0] = LAYOUT_68_ansi(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_MUTE,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_HOME,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,   MO(1), KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT_68_ansi(
```
Comment (dunk2k) [inline keyboards/phentech/rpk_001/keymaps/default/keymap.c:29]:
```suggestion
    [0] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_MUTE,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_HOME,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,   MO(1), KC_LEFT, KC_DOWN, KC_RGHT
    ),

    [1] = LAYOUT(
```

File: keyboards/phentech/rpk_001/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    [0] = LAYOUT_68_ansi(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_MUTE,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_PGDN,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,   MO(1), KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT_68_ansi(
```
Comment (dunk2k) [inline keyboards/phentech/rpk_001/keymaps/via/keymap.c:29]:
```suggestion
    [0] = LAYOUT(
        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_MUTE,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RCTL,   MO(1), KC_LEFT, KC_DOWN, KC_RGHT
    ),

    [1] = LAYOUT(
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/phentech/rpk_001/config.h [hunk 14]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 JoyLee (@itarze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* LED Driver */
```
Comment (dunk2k) [inline keyboards/phentech/rpk_001/config.h:6]:
```suggestion
/* RGB Config */
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/phentech/rpk_001/info.json [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+            {"pin_a": "A2", "pin_b": "A1"}
+        ]
+    },
+    "indicators": {
+        "caps_lock": "C15",
+        "scroll_lock": "C0",
+        "on_state": 0
+    },
+    "rgb_matrix": {
+        "driver": "AW20216",
```
Comment (itarze) [inline keyboards/phentech/rpk_001/info.json:38]:
```suggestion
        "driver": "aw20216",
```

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**